platform/ios/src/MGLMapView.mm

```diff
@@ -4759,7 +4804,11 @@ - (void)cameraIsChanging {
     }
 
     [self updateCompass];
-
+    
+    if (!self.scaleBarView.hidden) {
+        [(MGLScaleBarView *)self.scaleBarView setMetersPerPoint:[self metersPerPointAtLatitude:self.centerCoordinate.latitude]];
```

Comment:
If the zoom interaction occurred less than one second ago, it would show the incorrect scale for the remaining milliseconds up to 1 sec but since it was hidden, we never call `-fadeIn` leaving the view.alpha at 0 making the incorrect scale invisible. You have to set `metersPerPoint` again to make it visible.

Though, it is possible to trigger the case you think of but it's slightly harder in practice;
Using `scaleView.hidden = NO`, zoom to fade in the scale bar, hide the scale bar, change the zoom, and unhide the scale bar all within 1 second in order for it to show an incorrect scale.

However, we can remove the if-statement if the overhead is exaggerated.